src/screens/PinScreen.js

```diff
@@ -161,11 +170,26 @@ class PinScreen extends React.Component {
     this.setState({ pin: text, pinColor: 'black', error: null });
   }
 
-  validatePin = (text) => {
-    if (hathorLib.wallet.isPinCorrect(text)) {
-      this.dismiss(text);
-    } else {
-      this.removeOneChar();
+  validatePin = (pin) => {
+    try {
+      if (hathorLib.wallet.isPinCorrect(pin)) {
+        // also validate if we are able to decrypt the seed using this PIN
+        // this will throw if the words are not able to be decoded with this
+        // pin.
+        hathorLib.wallet.getWalletWords(pin);
+
+        if (!hathorLib.wallet.wordsValid(pin)) {
+          throw new Error('Words decrypted with pin are invalid');
+        }
+
+        this.dismiss(pin);
+      } else {
+        this.removeOneChar();
+      }
+    } catch (e) {
+      this.props.unlockScreen();
+      this.props.setRecoveringPin(true);
```

Comment:
Can't we send the temporary pin to this method (and remove the next call)?

Reply:
We could, but I think that having those two actions being dispatched make it more explicit and easier to read, without compromising performance, so I see no downside of keeping both

But I can refactor it if you prefer having a single action

Reply:
I'm mostly concerning on misusage of these actions. But a minor change can make it safe enough for me. I commented about it in another conversation.